File: src/app/shell/XXXXX.Shell.Core/Interfaces/IRolePermissionService.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;

using XXXXX.Shell.Core.ViewModels;

namespace XXXXX.Shell.Core.Interfaces {
    public interface IRolePermissionService {
        Task<IEnumerable<PermissionInfosViewModel>> GetMany(RequestHeaders headers, Guid organisationId, Guid roleId);
        Task Update(RequestHeaders headers, Guid organisationId, Guid roleId, List<Guid> payload);
    }
}

File: src/app/shell/XXXXX.Shell.UI/src/interfaces/irolePermissionService.ts
import { PermissionInfos, PermissionsFilter } from "@/domain/models/permissions";

export interface IRolePermissionService {
    getMany(organisationId: string, roleId: string, filter?: PermissionsFilter): Promise<PermissionInfos[]>;
    update(organisationId: string, roleId: string, payload: string[]): Promise<void>;
}